extensions/amp-selector/1.0/README.md

```diff
@@ -0,0 +1,287 @@
+  </li>
+</ul>
+
+```js
+selector.addEventListener("select", (e) => console.log(e.data.targetOption))
+```
+
+### Preact/React Component
+
+The examples below demonstrate use of the `<BentoSelector>` as a functional component usable with the Preact or React libraries.
+
+#### Example: Import via npm
+
+[example preview="top-frame" playground="false"]
+
+Install via npm:
+
+```sh
+npm install @ampproject/bento-selector
+```
+
+```javascript
+import React from 'react';
+import { BentoSelector, BentoSelectorOption} from '@ampproject/bento-selector/react';
+import '@ampproject/bento-selector/styles.css';
+
+function App() {
+  return (
+    <BentoSelector aria-label="Option menu">
+      <BentoSelectorOption option="1">Option 1</BentoSelectorOption>
+      <BentoSelectorOption option="2">Option 2</BentoSelectorOption>
+      <BentoSelectorOption option="3">Option 3</BentoSelectorOption>
+      <BentoSelectorOption option="4">Option 4</BentoSelectorOption>
+    </BentoSelector>
+  );
+}
+```
+
+[/example]
+
+#### Layout and Style
+
+Each Bento component has a small CSS library you must include to guarantee proper loading without [content shifts](https://web.dev/cls/). Because of order-based specificity, you must manually ensure that stylesheets are included before any custom styles.
+
+```html
+<link rel="stylesheet" type="text/css" href="https://cdn.ampproject.org/v0/amp-facebook-1.0.css">
+```
+
+Alternatively, you may also make the light-weight pre-upgrade styles available inline:
+
+```html
+<style data-bento-boilerplate>
+ BentoSelector{
+    display: block;
+    overflow: hidden;
+    position: relative;
+  }
+</style>
+```
+
+#### Attributes on `<BentoSelector>`
+
+**disabled, form, multiple, name**
+
+The attributes above behave the same way as they do on a standard HTML [`<select>`](https://developer.mozilla.org/en/docs/Web/HTML/Element/select) element.
+
+**keyboard-select-mode**
+
+The `keyboard-select-mode` attribute dictates the keyboard navigation behavior for options inside `<BentoSelector>`.
+
+<ul>
+  <li>
+    `none` (default): The tab key changes focus between items in the `<BentoSelector>`. The user must press enter or space to change the selection. Arrow keys are disabled.
+  </li>
+  <li>
+    `focus`: Tab key gives focus to `<BentoSelector>`. The user navigates between items with the arrow keys. Must press space or enter to change the selection.
+  </li>
+  <li>
+    `select`: Tab key gives focus to `<BentoSelector>`. The selection changes as the user navigates options with arrow keys.
+  </li>
+</ul>
+
+#### Attributes on `<BentoSelectorOption>`
+
+**option**
+
+Indicates that the option is selectable. If a value is specified, the contents of the value is submitted with the form.
+
+**disabled, selected**
+
+The attributes above behave the same way as they do on a standard HTML [`<option>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option) element.
+
+#### Actions
```

Comment:
Please add an example for how to pass the ref and use the api on the ref, like in other Preact README files. Here's the section from sidebar:

```md
#### Interactivity and API usage

Bento components are highly interactive through their API. The `BentoSidebar` component API is accessible by passing a `ref`:

``javascript
import React, {createRef} from 'react';
const ref = createRef();

function App() {
  return (
    <BentoSidebar ref={ref}>
      <ul>
        <li>Nav item 1</li>
        <li>Nav item 2</li>
        <li>Nav item 3</li>
        <li>Nav item 4</li>
        <li>Nav item 5</li>
        <li>Nav item 6</li>
      </ul>
    </BentoSidebar>
  );
}
``

##### Actions

The `BentoSidebar` API allows you to perform the following actions:

**open()**
Opens the sidebar.

``javascript
ref.current..open();
``

**close()**
Closes the sidebar.

``javascript
ref.current..close();
``

**toggle()**
Toggles the sidebar open state.

``javascript
ref.current..toggle(0);
``
```

Note: Please update the double ticks to triple ticks for the code snippets above -- I removed them here so they would all be shown in one markdown block in the GH formatting.